pysmartmeter/publish_loop.py

```diff
@@ -90,15 +94,36 @@ def __call__(self, *, obis_values: list[ObisValue]):
 
 
 def publish_forever(*, settings: MqttSettings, verbose):
-    serial = get_serial()
+    import serial
+    config_manager = ConfigManager()
+
+    if config_manager.BINARY_MODE:
```

Comment:
These kind of "is config XY is ensable" looks very non-common. Please take a look into pep-8: https://peps.python.org/pep-0008/